### PR TITLE
First step of refactoring routes

### PR DESCRIPTION
**Pull request by GarySchwab**
Currently only one function handles all the post request, the objective is to split that function into several smaller ones.

### CODE REVIEW

**Comment by GarySchwab** [comment]
I accidentally changed the indentation size of post.ts, that's why the whole file changed.

File: app/routes/post.ts [hunk 1]
```diff
@@ -19,501 +19,477 @@ import cloudinary from "cloudinary";
...
+	) {
+		const url = await uploadPhoto(photo);
+		updateUser(id_user, { photo: url });
+	}
+	
+	/**
+	 * This function uploads a photo and returns its url
+	 * @param photo base64 image
+	 */
+	function uploadPhoto(photo) {
```
Comment (ayshiff) [inline app/routes/post.ts]: We should be able to do this instead : 
```
function uploadPhoto(photo) {
		return cloudinary.uploader
			.upload("data:image/jpeg;base64," + photo)
			.then(result => result.secure_url)
			.catch(error => console.log(error));
	}
```
And by adding what it should return on the top :
```
	/**
	 * This function uploads a photo and returns its url
	 * @param photo base64 image
         * @returns the uploaded image url
	 */
```

File: app/routes/post.ts [hunk 2]
```diff
@@ -19,501 +19,477 @@ import cloudinary from "cloudinary";
 
 const HTTPS_REGEX = "^https?://(.*)";
 
+const errorMessages = {
```
Comment (ayshiff) [inline app/routes/post.ts]: 👍

File: app/routes/post.ts [hunk 3]
```diff
@@ -19,501 +19,477 @@ import cloudinary from "cloudinary";
...
+	/**
+	 * This function is used to know if a place exists and who uses it.
+	 * @param {string} id_place id of the current place
+	 */
+	async function whoUses(id_place: string) {
+		return await new Promise((resolve, reject) => {
+			Place.findOne({ id: id_place }, (err: Error, place: PlaceSchema) => {
+				if (err) RES.status(resultCodes.serverError).send(errorMessages.placeFind);
+				else if (place !== null) resolve(place.id_user); // "" => not used, "NAME" => used by NAME
+				else resolve("#"); // place not exists
```
[the remark this answers is not in the document]
Comment (GarySchwab) [reply app/routes/post.ts]: This will be done in another pull request, as the # symbol resolved by this function is used in other functions too.

**Pull request merged**